packages/lib/src/SingleSignOn.shared.ts

```diff
@@ -1,93 +1,187 @@
+      JSON.parse(identityStr);
+
+    Validations.validateAuthIdentity(identityWithStringExpiration);
+
+    const identity: AuthIdentity = {
+      ...identityWithStringExpiration,
+      expiration: new Date(identityWithStringExpiration.expiration),
+    };
+
+    identity.expiration = new Date(identity.expiration);
+
+    if (identity.expiration.getTime() <= Date.now()) {
+      return null;
+    }
+
+    return identity;
   }
 
-  let identity: AuthIdentity;
-  try {
-    // If the item is not a valid JSON, we remove it.
-    identity = JSON.parse(item);
-  } catch (e) {
-    localStorageClearIdentity(user);
-    return null;
+  export function setIdentity(address: string, identity: AuthIdentity | null): void {
+    Validations.validateAddress(address);
+
+    const key = getIdentityKey(address);
+
+    if (!identity) {
+      localStorage.removeItem(key);
+    } else {
+      Validations.validateAuthIdentity(JSON.parse(JSON.stringify(identity)));
+
+      localStorage.setItem(key, JSON.stringify(identity));
+    }
   }
 
-  // The expiration is parsed as a string, so we need to convert it to a Date.
-  identity.expiration = new Date(identity.expiration);
+  export function getConnectionData(): ConnectionData | null {
+    const connectionDataStr = localStorage.getItem(CONNECTION_DATA_KEY);
 
-  // If expiration is in the past, we remove the identity.
-  if (identity.expiration.getTime() <= Date.now()) {
-    localStorageClearIdentity(user);
-    return null
+    if (!connectionDataStr) {
+      return null;
+    }
+
+    const connectionData = JSON.parse(connectionDataStr) as ConnectionData;
+
+    Validations.validateConnectionData(connectionData);
+
+    return connectionData as ConnectionData;
   }
 
-  // Everything is fine, we return the identity.
-  return identity;
-}
+  export function setConnectionData(connectionData: ConnectionData | null): void {
+    if (!connectionData) {
+      localStorage.removeItem(CONNECTION_DATA_KEY);
+    } else {
+      Validations.validateConnectionData(connectionData);
 
-// Stores the identity of the user in local storage (if is not expired).
-export function localStorageStoreIdentity(user: string, identity: AuthIdentity) {
-  const expiration = new Date(identity.expiration)
-  if (expiration.getTime() > Date.now()) {
-    localStorage.setItem(getKey(user), JSON.stringify(identity));
+      localStorage.setItem(CONNECTION_DATA_KEY, JSON.stringify(connectionData));
+    }
   }
-}
 
-// Clears the identity of the user from local storage.
-export function localStorageClearIdentity(user: string) {
-  localStorage.removeItem(getKey(user));
+  function getIdentityKey(address: string) {
+    return `${IDENTITY_KEY}-${address}`;
+  }
 }
 
-// Gets the key where the identity of the user is stored in local storage.
-function getKey(user: string) {
-  if (!/^0x[a-fA-F0-9]{40}$/.test(user)) {
-    throw new Error(`User must be a valid ethereum address`);
+namespace Validations {
```

Comment:
I would recommend you to no use namespaces as [they're not recommended by the typescript documentation](https://www.typescriptlang.org/docs/handbook/namespaces-and-modules.html#using-modules). Namespaces can be replaced by a module.

Reply:
The pitfalls described in the docs don't seem to relevant. And namespaces are used for most dcl schemas types. Besides, it makes it easier to export a single shared file instead of having to import and export things from different places.